packages/api/endpoints/rules.js

```diff
@@ -60,26 +61,28 @@ async function get(req, res) {
  * @returns {Promise<Object>} the promise of express response object
  */
 async function post(req, res) {
+  const {
+    model = new models.Rule(),
+    dbClient = await getKnexClient(),
+  } = req.testContext || {};
+
+  const rule = req.body || {};
+  const name = rule.name;
+
+  if (await model.exists(name)) {
+    return res.boom.conflict(`A record already exists for ${name}`);
+  }
+
   try {
-    const data = req.body;
-    const name = data.name;
-
-    const model = new models.Rule();
-
-    try {
-      await model.get({ name });
-      return res.boom.conflict(`A record already exists for ${name}`);
-    } catch (error) {
-      if (error instanceof RecordDoesNotExist) {
-        const record = await model.create(data);
-
-        if (inTestMode()) {
-          await addToLocalES(record, indexRule);
-        }
-        return res.send({ message: 'Record saved', record });
-      }
-      throw error;
-    }
+    return await dbClient.transaction(async (trx) => {
+      const record = await model.create(rule);
+      const postgresRecord = await translateApiRuleToPostgresRule(record, dbClient);
+      await trx(tableNames.rules).insert(postgresRecord, 'cumulus_id');
+
+      if (inTestMode()) await addToLocalES(record, indexRule);
+
+      return res.send({ message: 'Record saved', record });
```

Comment:
I don't think we want either of these things to be a part of the transaction, so they should come after. They are not related to the logic of writing to our databases